### PR TITLE
fix: allow plugin instance (#5760)

### LINKED ISSUE #5760: plugins options support plugin instance too
**Environments:**
- Prettier Version: 1.15.3
- Running Prettier via: 
- Runtime: 
- Operating System: 

**Steps to reproduce:**

I wanna to build a app target web(by standalone and external modules) and node,
I tried to donot explicit set plugin by explicit behavior in my code, 
but the plugins option of standalone accept plugin instance, the otherwise accept a path,
Could we accept the plugin instacne too in the normal version? that maybe only a typeof pred statement but very helpful to me!

**Expected behavior:**

**Actual behavior:**

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #5760 

- [ ] I’ve added tests to confirm my change works. ?
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by Kingwl** [comment]
#5760

**Comment by evilebottnawi** [comment]
Need tests

**Comment by Kingwl** [comment]
I am not sure how to add this test.

**Comment by evilebottnawi** [comment]
@Kingwl you can find test's examples here https://github.com/prettier/prettier/tree/master/tests_integration/__tests__

File: src/common/load-plugins.js [hunk 1]
```diff
@@ -24,7 +24,19 @@ function loadPlugins(plugins, pluginSearchDirs) {
...
+    ([instances, names], next) => {
+      if (typeof next === "string") {
+        return [instances, names.concat([next])];
+      }
+      return [instances.concat([next]), names];
+    },
+    [[], []]
+  );
+  const externalPluginInstances = pluginResult[0];
+  const externalPluginNames = pluginResult[1];
```
[the remark this answers is not in the document]
Comment (Kingwl) [reply src/common/load-plugins.js]: Actually, the test env in ci whitch node4 does not support array destruction

[the remark this answers is not in the document]
Comment (ikatyang) [reply src/common/load-plugins.js]: The node v4 error is coming from yarn (yarnpkg/yarn#6900, see #5766), we do compile our code to node v4 before testing.

[the remark this answers is not in the document]
Comment (Kingwl) [reply src/common/load-plugins.js]: should partition extract to a common file like utils?

[the remark this answers is not in the document]
Comment (ikatyang) [reply src/common/load-plugins.js]: 👍

**Pull request merged**